services/piwheels/piwheels-version.service.js

```diff
@@ -0,0 +1,94 @@
+import Joi from 'joi'
+import { BaseJsonService, InvalidResponse } from '../index.js'
+import { renderVersionBadge } from '../version.js'
+
+const schema = Joi.object({
+  releases: Joi.object()
+    .pattern(
+      Joi.string(),
+      Joi.object({
+        prerelease: Joi.boolean().required(),
+        yanked: Joi.boolean().required(),
+      })
+    )
+    .required(),
+}).required()
+
+const queryParamSchema = Joi.object({
+  include_prereleases: Joi.equal(''),
+}).required()
+
+const keywords = ['python', 'arm', 'raspberry pi']
+
+export default class PiWheelsVersion extends BaseJsonService {
+  static category = 'version'
+
+  static route = { base: 'piwheels/v', pattern: ':wheel', queryParamSchema }
+
+  static examples = [
+    {
+      title: 'piwheels',
+      namedParams: { wheel: 'numpy' },
+      staticPreview: this.render({ version: '1.22.2' }),
+      keywords,
+    },
+    {
+      title: 'piwheels (including prereleases)',
+      namedParams: { wheel: 'flask' },
+      queryParams: {
+        include_prereleases: null,
+      },
+      staticPreview: this.render({ version: '2.0.0rc2' }),
+      keywords,
+    },
+  ]
+
+  static defaultBadgeData = { label: 'piwheels' }
+
+  static render({ version }) {
+    return renderVersionBadge({ version })
+  }
+
+  async fetch({ wheel }) {
+    return this._requestJson({
+      schema,
+      url: `https://www.piwheels.org/project/${wheel}/json/`,
+      errorMessages: { 404: 'package not found' },
+    })
+  }
+
+  static transform(releases, includePrereleases) {
+    const allReleases = Object.keys(releases)
+      .reduce(
+        (acc, key) =>
+          acc.concat({
+            version: key,
+            prerelease: releases[key].prerelease,
+            yanked: releases[key].yanked,
+          }),
+        []
+      )
+      .filter(release => !release.yanked) // exclude any yanked releases
+
+    if (allReleases.length === 0) {
+      throw new InvalidResponse({ prettyMessage: 'no versions found' })
+    }
+
+    if (includePrereleases) {
+      return allReleases[0].version
+    }
```

Comment:
I didn't see anything in the docs indicating a sort order, so figured I'd double check to confirm they are?

Reply:
Yeah its a good question. I did look over the source for piwheels. The code that generates the HTML pages and the JSON API does parse all the versions to PEP440 version objects and then sorts them. https://github.com/piwheels/piwheels/blob/fd6f3d12d9a33390ccb235a2995edb24a8fa782d/piwheels/master/the_scribe.py#L525-L542 I'm happy to rely on the ordering, even though it is not explicitly called out in the API docs.